database/aggregate.js

```diff
@@ -10,17 +10,27 @@ const generateAggregate = product => new Promise((resolve) => {
     if (err) throw err;
     const aggregates = {};
     let total = 0;
+    let five = 0;
+    let four = 0;
+    let three = 0;
+    let two = 0;
+    let one = 0;
     data.forEach((rating) => {
       total += rating.rating;
+      if (rating.rating === 5) five += 1;
+      if (rating.rating === 4) four += 1;
+      if (rating.rating === 3) three += 1;
+      if (rating.rating === 2) two += 1;
+      if (rating.rating === 1) one += 1;
```

Comment:
You might be able to store these ratings in an array, and use the star rating as a reference to the array's index. Increase the array value by 1.

Reply:
Smart! Ticket added to Trello for optimization when time permits